src/DynamoCoreWpf/ViewModels/Search/SearchViewModel.cs

```diff
@@ -1061,14 +1073,56 @@ public void OnSearchElementClicked(NodeModel nodeModel, Point position)
 
         internal void OnRequestConnectToPort(string nodeCreationName, PortModel portModel)
         {
-            if (!nodeCreationName.Contains("Code Block"))
+            // Do not auto connect code block node since default code block node do not have output port
+            if (nodeCreationName.Contains("Code Block")) return;
+            
+            var initialNode = dynamoViewModel.CurrentSpaceViewModel.Nodes.FirstOrDefault(x => x.Id == portModel.Owner.GUID);
+            var id = Guid.NewGuid();
+
+            var adjustedX = initialNode.X;
+
+            // Placing the new node based on which input port it is connecting to.
+            if (portModel.PortType == PortType.Input)
             {
-                // Create a new node based on node creation name and connect ports
-                dynamoViewModel.ExecuteCommand(new DynamoModel.CreateAndConnectNodeCommand(Guid.NewGuid(), portModel.Owner.GUID,
-                    nodeCreationName, 0, portModel.Index, portModel.CenterX - 50, portModel.CenterY + 200, false, false));
+                // Placing the new node to the left of initial node
+                adjustedX -= portModel.Owner.Width + 50;
+            }
+            else
+            {
+                adjustedX += portModel.Owner.Width + 50;
             }
 
-            OnRequestFocusSearch();
+            if (undoRecorderGroup == null)
+            {
+                dynamoViewModel.NodeViewReady += AutoLayoutNodes;
+                undoRecorderGroup = dynamoViewModel.CurrentSpace.UndoRecorder.BeginActionGroup();
+            }
+
+            // Create a new node based on node creation name and connect ports
+            dynamoViewModel.ExecuteCommand(new DynamoModel.CreateAndConnectNodeCommand(id, portModel.Owner.GUID,
+                nodeCreationName, 0, portModel.Index, adjustedX, 0, false, false, true));
+
+            // Clear current selections and select all input nodes
+            DynamoSelection.Instance.ClearSelection();
+            var inputNodes = portModel.Owner.InputNodes.Values.Where(x => x != null).Select(y => y.Item2);
```

Comment:
Can you explain the Item2 part? I think I tried this one before by just calling Select and did not work.. Maybe i missed something

Reply:
Haha, nothing smart here. `InputNodes` property on `NodeModel` is of type `IDictionary<int, Tuple<int, NodeModel>>`. I'm just getting the `NodeModel` part of the tuple.